elastalert/elastalert.py

```diff
@@ -719,6 +721,39 @@ def start(self):
             sleep_duration = (next_run - datetime.datetime.utcnow()).seconds
             self.sleep_for(sleep_duration)
 
+    def execute_rule(self, rule):
+        next_run = datetime.datetime.utcnow() + self.run_every
+
+        # Set endtime based on the rule's delay
+        delay = rule.get('query_delay')
+        if hasattr(self.args, 'end') and self.args.end:
+            endtime = ts_to_dt(self.args.end)
+        elif delay:
+            endtime = ts_now() - delay
+        else:
+            endtime = ts_now()
+
+        try:
+            num_matches = self.run_rule(rule, endtime, self.starttime)
+        except EAException as e:
+            self.handle_error("Error running rule %s: %s" % (rule['name'], e), {'rule': rule['name']})
+        except Exception as e:
+            self.handle_uncaught_exception(e, rule)
+        else:
+            old_starttime = pretty_ts(rule.get('original_starttime'), rule.get('use_local_time'))
+            elastalert_logger.info("Ran %s from %s to %s: %s query hits, %s matches,"
+                                   " %s alerts sent" % (rule['name'], old_starttime, pretty_ts(endtime, rule.get('use_local_time')),
+                                                        self.num_hits, num_matches, self.alerts_sent))
```

Comment:
self.num_hits needs to be mapped per rule. (It should have been in the first place!)